### PR TITLE
Analog wrapper harmonize params

### PR DESCRIPTION
**Pull request by barbalberto**
-Change ROS param from `joint_names` to `jointNames` like the controlBoardWrapper. Old name is deprecated, but still accepted. A warning message will be displayed.

- Add call to subdevice close

### CODE REVIEW

File: src/libYARP_dev/src/devices/AnalogWrapper/AnalogWrapper.cpp [hunk 1]
```diff
@@ -848,11 +880,16 @@ bool AnalogWrapper::close()
...
     }
 
-    //RateThread::stop();
-
     detachAll();
     removeHandlers();
 
+    if(subDeviceOwned)
+    {
+        yDebug() << "free subdevice";
```
Comment (traversaro) [inline src/libYARP_dev/src/devices/AnalogWrapper/AnalogWrapper.cpp]: Spurious?

**Comment by traversaro** [comment]
This may be a more general comment, but here it is: when debugging the output of the yarprobotinterface, it is a lot of time confusing when you get a warning/error without any information about which device printed the error.. could it make sense to prepend all the warning/error outputs with "AnalogWrapper: "?

**Comment by barbalberto** [comment]
Yes, all the print of this device start with
`  yError() << sensorId << "message";`

Also in the   CBW  I just merged a change which propagates the name of the subdevice to improve those messages

**Comment by traversaro** [comment]
Ah, I see, I missed it because I believed that `deviceId` was something else, thanks!

File: src/libYARP_dev/src/devices/AnalogWrapper/AnalogWrapper.cpp [hunk 2]
```diff
@@ -848,11 +880,15 @@ bool AnalogWrapper::close()
...
 
-    //RateThread::stop();
-
     detachAll();
     removeHandlers();
 
+    if(subDeviceOwned)
+    {
+        subDeviceOwned->close();
+        delete subDeviceOwned;
```
Comment (traversaro) [inline src/libYARP_dev/src/devices/AnalogWrapper/AnalogWrapper.cpp]: Not fundamental, but it is always good practice to set to `YARP_NULLPTR` a pointer after the delete.

Comment (barbalberto) [inline src/libYARP_dev/src/devices/AnalogWrapper/AnalogWrapper.cpp]: Right! Done

**Pull request merged**